eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -58,24 +58,24 @@ object OnionMessages {
         .map(tlvs => RouteBlindingEncryptedDataCodecs.blindedRouteDataCodec.encode(TlvStream(tlvs)).require.bytes)
     }
     destination match {
-      case BlindedPath(route) if originKey.exists(route.introductionNodeId == _.publicKey) =>
-        val (nextNodeId, nextBlinding) = RouteBlindingEncryptedDataCodecs.decode(originKey.get, route.blindingKey, route.blindedNodes.head.encryptedPayload) match {
-          case Left(invalidData) => throw new Exception(invalidData.toString)
-          case Right(decoded) =>
-            (decoded.tlvs.get[RouteBlindingEncryptedDataTlv.OutgoingNodeId].get.nodeId,
-              decoded.tlvs.get[RouteBlindingEncryptedDataTlv.NextBlinding].map(_.blinding).getOrElse(decoded.nextBlinding))
+      case BlindedPath(route) if nodeKey.publicKey == route.introductionNodeId =>
+        RouteBlindingEncryptedDataCodecs.decode(nodeKey, route.blindingKey, route.blindedNodes.head.encryptedPayload) match {
+          case Left(f) => Failure(new IllegalArgumentException(f.message))
+          case Right(decoded) => decoded.tlvs.get[RouteBlindingEncryptedDataTlv.OutgoingNodeId].map(_.nodeId) match {
+            case Some(nextNodeId) => Success(Sphinx.RouteBlinding.BlindedRoute(nextNodeId, decoded.nextBlinding, route.blindedNodes.tail))
```

Comment:
What happened to `decoded.tlvs.get[RouteBlindingEncryptedDataTlv.NextBlinding]`? If it is present, it should overwrite `decoded.nextBlinding`.

Reply:
It was just duplicating the code that is inside `RouteBlindingEncryptedDataCodecs.decode`.

Reply:
good catch, thanks